src/server/api/routes/users/v2/notification.js

```diff
@@ -0,0 +1,64 @@
+const { Router } = require('express');
+const {
+  authenticateV2Login,
+  checkEmailHasVerified,
+} = require('../../../middleware/auth');
+const { setPrivateCacheHeader } = require('../../../middleware/cache');
+const { handleRestfulError } = require('../../../middleware/error');
+const { walletUserCollection } = require('../../../../modules/firebase');
+
+const router = Router();
+
+router.get('/notification', authenticateV2Login, async (req, res, next) => {
+  try {
+    const { user } = req.session;
+    const userDoc = await walletUserCollection.doc(user).get();
+    const { notification = {} } = userDoc.data();
+    const { transfer = true, purchasePrice = 0 } = notification;
+    res.json({
+      notification: {
+        transfer,
+        purchasePrice,
+      },
+    });
+  } catch (err) {
+    handleRestfulError(req, res, next, err);
+  }
+});
+
+router.post(
+  '/notification',
+  authenticateV2Login,
+  checkEmailHasVerified,
+  async (req, res, next) => {
+    try {
+      setPrivateCacheHeader(res);
```

Comment:
We should set private cache header in `authenticateV2Login`? Both security and code styling sense it seems to be an improvement
Also setting private cache for POST seems redundant